src/bp/nlu-core/entities/microsoft-extractor/typings.ts

```diff
@@ -0,0 +1,219 @@
+import Recognizers from '@microsoft/recognizers-text-suite'
+
+export interface MicrosoftValue {
+  value: string
+  unit?: string
+  type?: string
+  score?: number
+  otherResults?: any[]
+}
+
+export interface MicrosoftTimeValues {
+  timex: string
+  type: string
+  start?: string
+  end?: string
+  value?: string
+  Mod?: string
+  sourceEntity?: string
+}
+
+export interface MicrosoftValues {
+  values: MicrosoftTimeValues[]
+}
+
+export type MicrosoftResolution = MicrosoftValue | MicrosoftValues
+
+export interface MicrosoftEntity {
+  start: number
+  end: number
+  resolution: MicrosoftResolution
+  text: string
+  typeName: string
+}
+export type SupportedLangs = 'zh' | 'nl' | 'en' | 'fr' | 'de' | 'it' | 'ja' | 'pt' | 'es'
+export const SupportedLangsList = ['zh', 'nl', 'en', 'fr', 'de', 'it', 'ja', 'pt', 'es']
+
+export const langToCulture = (lang: SupportedLangs): string | undefined => {
+  switch (lang) {
+    case 'zh':
+      return Recognizers.Culture.Chinese
+    case 'nl':
+      return Recognizers.Culture.Dutch
+    case 'en':
+      return Recognizers.Culture.English
+    case 'fr':
+      return Recognizers.Culture.French
+    case 'de':
+      return Recognizers.Culture.German
+    case 'it':
+      return Recognizers.Culture.Italian
+    case 'ja':
+      return Recognizers.Culture.Japanese
+    case 'pt':
+      return Recognizers.Culture.Portuguese
+    case 'es':
+      return Recognizers.Culture.Spanish
+    default:
+      return undefined
```

Comment:
better  👍 

\+ I saw there's a check for undefined culture right after calling this function